server/container/core/src/main/java/org/apache/james/server/core/MailImpl.java

```diff
@@ -427,28 +418,16 @@ public MailImpl(String name, MailAddress sender, Collection<MailAddress> recipie
         }
     }
 
-    @SuppressWarnings({"unchecked", "deprecated"})
+    @SuppressWarnings({"deprecation"})
     private MailImpl(Mail mail, String newName) throws MessagingException {
         this(newName, mail.getSender(), mail.getRecipients(), mail.getMessage());
         setRemoteHost(mail.getRemoteHost());
         setRemoteAddr(mail.getRemoteAddr());
         setLastUpdated(mail.getLastUpdated());
         setErrorMessage(mail.getErrorMessage());
-        try {
-            if (mail instanceof MailImpl) {
-                setAttributesRaw((Map<String, Object>) cloneSerializableObject(((MailImpl) mail).getAttributesRaw()));
-            } else {
-                HashMap<String, Object> attribs = new HashMap<>();
-                for (Iterator<String> i = mail.getAttributeNames(); i.hasNext(); ) {
-                    String hashKey = i.next();
-                    attribs.put(hashKey, cloneSerializableObject(mail.getAttribute(hashKey)));
-                }
-                setAttributesRaw(attribs);
-            }
-        } catch (IOException | ClassNotFoundException e) {
-            LOGGER.error("Error while deserializing attributes", e);
-            setAttributesRaw(new HashMap<>());
-        }
+        this.attributes = mail.attributes()
+            .map(attribute -> attribute.duplicate())
```

Comment:
Why is duplication needed?

Isn't `Attribute` class immutable?

Reply:
Yes, `AttributeName` and `AttributeValue` too, but  `AttributeValue`'s value can be mutable.